tools/autograd/gen_inplace_or_view_type.py

```diff
@@ -301,9 +308,8 @@ def emit_view_body(fn: NativeFunctionWithDifferentiabilityInfo, var: str) -> Tup
         else:
             _, unpacked_bindings = unpack_args(f)
             call += emit_view_lambda(f, unpacked_bindings)
-            creation_meta = ('at::GradMode::is_enabled() ? '
-                             'CreationMeta::DEFAULT : '
-                             'CreationMeta::NO_GRAD_MODE')
+            creation_meta = ('(!at::GradMode::is_enabled() || InferenceMode::is_enabled()) ? '
+                             'CreationMeta::NO_GRAD_FN : CreationMeta::DEFAULT')
```

Comment:
Is `NO_GRAD_FN` really that accurate a new name for the CreationMeta? To me, this reads like "this as no `grad_fn`", but I don't think the CreationMeta here has any thing to do with whether or not you have a grad fn; for example, if I have a `requires_grad=False` tensor and I take a view on it, IIUC, I'll get a view with `CreationMeta::DEFAULT`, even though the `grad_fn` is empty.